Hunting Queries/Microsoft 365 Defender/Ransomware/ShadowCopyDeletion.yml

```diff
@@ -0,0 +1,43 @@
+id: ff9a1c12-e448-55ff-a576-7f53caa32044
+name: Shadow Copy Deletions
+description: |
+  This rule detects when Shadow Copies are being deleted. This is a know actions that is performed by TA.
+  This query detects know commands that have been used by the ransomware actors.
+  Some information from Mitre Attack: 
+  https://attack.mitre.org/techniques/T1490/
+requiredDataConnectors:
+- connectorId: MicrosoftThreatProtection
+  dataTypes:
+  - DeviceProcessEvents
+tactics:
+- Ransomware
+relevantTechniques:
+  - T1490
+query: |
+  let CommonRansomwareExecutionCommands = dynamic([@'vssadmin.exe delete shadows /all /quiet', 
+  @'wmic.exe shadowcopy delete', @'wbadmin delete catalog -quiet', 
+  @'Get-WmiObject Win32_Shadowcopy | ForEach-Object {$_.Delete();}',
+  @'del /s /f /q c:\*.VHD c:\*.bac c:\*.bak c:\*.wbcat c:\*.bkf c:\Backup*.* c:\backup*.* c:\*.set c:\*.win c:\*.dsk', 
+  @'wbadmin delete systemstatebackup -keepVersions:0', 
+  @'schtasks.exe /Change /TN "\Microsoft\Windows\SystemRestore\SR" /disable', 
+  @'schtasks.exe /Change /TN "\Microsoft\Windows\SystemRestore\SR" /enable >nul 2>&1', 
+  @'reg add "HKLM\SOFTWARE\Policies\Microsoft\Windows NT\SystemRestore" /v "DisableConfig" /t "REG_DWORD" /d "1" /f', 
+  @'reg add "HKLM\SOFTWARE\Policies\Microsoft\Windows NT\SystemRestore" /v "DisableSR" /t "REG_DWORD" /d "1" /f', 
+  @'reg add "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SystemRestore" /v "DisableConfig" /t "REG_DWORD" /d "1" /f', 
+  @'reg add "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SystemRestore" /v "DisableSR" /t "REG_DWORD" /d "1" /f', 
+  @'reg delete "HKLM\SOFTWARE\Policies\Microsoft\Windows NT\SystemRestore" /v "DisableConfig" /f >nul 2>&1', 
+  @'reg delete "HKLM\SOFTWARE\Policies\Microsoft\Windows NT\SystemRestore" /v "DisableSR" /f >nul 2>&1', 
+  @'reg delete "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SystemRestore" /v "DisableConfig" /f >nul 2>&1', 
+  @'reg delete "HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SystemRestore" /v "DisableSR" /f >nul 2>&1']);
+  DeviceProcessEvents
+  | where ProcessCommandLine has_any (CommonRansomwareExecutionCommands)
+  | project-reorder Timestamp, ProcessCommandLine, DeviceName, AccountName
```

Comment:
@Bert-JanP  -  Please change Timestamp to standard time column of TimeGenerated.  They are equal, but TimeGenerated is the standard.

Reply:
@Bert-JanP - Nevermind, missed this was in M365 Hunting folder, Timestamp is good here.